bsms.mediawiki

```diff
@@ -63,39 +63,39 @@ The Signer is a participating member in the multisig. Its responsibilities inclu
-* The Coordinator verifies that the included <tt>IV</tt> is valid given the decrypted data.
+* For each key record, the Coordinator extracts the <tt>MAC</tt> from the data, sets <tt>IV</tt> to <tt>MAC</tt>, then decrypts the ciphertext using <tt>ENCRYPTION_KEY</tt> and <tt>IV</tt>.
+* The Coordinator verifies that the included <tt>MAC</tt> is valid given the plaintext.
 * The Coordinator verifies that the included <tt>SIG</tt> is valid given the <tt>KEY</tt>.
 * If all key records look good, the Coordinator fills in all necessary information to generate a descriptor record, which is simply the descriptor string plus a <tt>CHECKSUM</tt>, all in one line. The <tt>CHECKSUM</tt> has [https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md#checksums BECH32 encoding].
-* The Coordinator encrypts the descriptor record with <tt>ENCRYPTION_KEY</tt>. The Coordinator calculates the 256-bit <tt>IV</tt> for the descriptor, and appends it to the end of the encrypted text.
+* The Coordinator calculates the <tt>MAC</tt> for the record. The <tt>MAC</tt> doubles as the <tt>IV</tt> for the encryption. 
+* The Coordinator encrypts the descriptor record with <tt>ENCRYPTION_KEY</tt> and <tt>IV</tt>. The Coordinator appends the MAC to the end of the encrypted text.
 * The Coordinator sends the encrypted descriptor record to all participating Signers.
 
 =====Signer=====
 
 * The Signer imports the descriptor record.
-* The Signer extracts the <tt>IV</tt> from the data, then decrypts the ciphertext using <tt>ENCRYPTION_KEY</tt> (derived from the open session) and <tt>IV</tt>.
-* The Signer verifies that the included <tt>IV</tt> is valid given the decrypted data.
-* The Signer calculates and verifies the descriptor’s <tt>CHECKSUM</tt>. Abort the setup if the <tt>CHECKSUM</tt> is incorrect.
-* Abort the setup if there are any <tt>KEY</tt> duplicates.
-* The Signer checks whether one of the <tt>KEY</tt>s in the descriptor belongs to it, using path and fingerprint information included in the descriptor. The check must perform an exact match on the <tt>KEY</tt>s, and not using shortcuts such as matching fingerprints (which is trivial to spoof). Abort the setup if it doesn’t detect its own <tt>KEY</tt>.
-* For confirmation, the Signer must display to the user the <tt>CHECKSUM</tt>, plus important policy parameters, such as <tt>M</tt>, <tt>N</tt> and the position of its own XPUB in the policy script. The total number of Signers, <tt>N</tt>, is important to prevent a <tt>KEY</tt> insertion attack. The position is important for scripts where key order matters. Except for the position, all participating Signers should be able to display the same confirmation.
+* The Signer extracts the <tt>MAC</tt> from the data, sets <tt>IV</tt> to <tt>MAC</tt>, then decrypts the ciphertext using <tt>ENCRYPTION_KEY</tt> (derived from the open session) and <tt>IV</tt>.
```

Comment:
`sets the <tt>IV</tt>`
`decrypts the ciphertext using the <tt>ENCRYPTION_KEY</tt> `